ssh_automated_tests/core/connection.py

```diff
@@ -19,12 +19,9 @@ def __init__(self, user: SshUser):
     def connect_via_key(self):
         print('connecting')
         connection_data = self.user.make_connection_data()
-        pkey = paramiko.RSAKey.from_private_key_file(connection_data["key"])
+        pkey = paramiko.RSAKey.from_private_key_file(connection_data.pop("key"))
         self.client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-        self.client.connect(hostname=connection_data["hostname"],
-                            username=connection_data["username"],
-                            port=connection_data["port"],
-                            pkey=pkey)
+        self.client.connect(**connection_data, pkey=pkey)
```

Comment:
> But the idea that the user don't know about paramiko or whatever else. He knows about his data and where his key is.

Yes, but using `dict['key']` you have to know a format of return. It means the user returns us a data structure instead of an object. 

Okay, then make `ParamikoUser` class which will provide connection details especially for Paramiko when we call `connection_data()`.